dist-raw/node-primordials.js

```diff
@@ -1,24 +1,427 @@
-module.exports = {
-  ArrayIsArray: Array.isArray,
-  ArrayPrototypeJoin: (obj, separator) => Array.prototype.join.call(obj, separator),
-  ArrayPrototypeShift: (obj) => Array.prototype.shift.call(obj),
-  ArrayPrototypeForEach: (arr, ...rest) => Array.prototype.forEach.apply(arr, rest),
-  JSONParse: JSON.parse,
-  JSONStringify: JSON.stringify,
-  ObjectFreeze: Object.freeze,
-  ObjectGetOwnPropertyNames: Object.getOwnPropertyNames,
-  ObjectDefineProperty: Object.defineProperty,
-  ObjectPrototypeHasOwnProperty: (obj, prop) => Object.prototype.hasOwnProperty.call(obj, prop),
-  RegExpPrototypeTest: (obj, string) => RegExp.prototype.test.call(obj, string),
-  SafeMap: Map,
-  SafeSet: Set,
-  StringPrototypeEndsWith: (str, ...rest) => String.prototype.endsWith.apply(str, rest),
-  StringPrototypeIncludes: (str, ...rest) => String.prototype.includes.apply(str, rest),
-  StringPrototypeLastIndexOf: (str, ...rest) => String.prototype.lastIndexOf.apply(str, rest),
-  StringPrototypeIndexOf: (str, ...rest) => String.prototype.indexOf.apply(str, rest),
-  StringPrototypeReplace: (str, ...rest) => String.prototype.replace.apply(str, rest),
-  StringPrototypeSlice: (str, ...rest) => String.prototype.slice.apply(str, rest),
-  StringPrototypeSplit: (str, ...rest) => String.prototype.split.apply(str, rest),
-  StringPrototypeStartsWith: (str, ...rest) => String.prototype.startsWith.apply(str, rest),
-  StringPrototypeSubstr: (str, ...rest) => String.prototype.substr.apply(str, rest),
+// copied from https://github.com/nodejs/node/blob/394fdecc9141aa4c6a8ea7f1aecaf5fcaaaf5c72/lib/internal/per_context/primordials.js
+'use strict';
+
+/* eslint-disable node-core/prefer-primordials */
+
+// This file subclasses and stores the JS builtins that come from the VM
+// so that Node.js's builtin modules do not need to later look these up from
+// the global proxy, which can be mutated by users.
+
+// Use of primordials have sometimes a dramatic impact on performance, please
+// benchmark all changes made in performance-sensitive areas of the codebase.
+// See: https://github.com/nodejs/node/pull/38248
+
+globalThis.primordials = {};
```

Comment:
Is it necessary to be leaking internal implementation details onto the global object?

Reply:
Node binding loader injects `primordials` into `globalThis`, but we're not using the loader, hence the need to make that adjustment (since it doesn't exists when running that file).

Reply:
Now I understand what you meant by that, and absolutely, we don't need to do this way as opposed to just exporting `primordials` object, I've fixed it.